bokehjs/src/coffee/models/glyphs/line.coffee

```diff
@@ -69,7 +69,7 @@ export class LineView extends XYGlyphView
       values = @_x
 
     for i in [0...values.length-1]
-      if values[i]<=val<=values[i+1]
+      if values[i]<=val<=values[i+1] or values[i+1]<=val<=values[i]
```

Comment:
You are correct that this change is necessary. Without it, it won't properly inspect lines going "backwards".

```python
from bokeh.plotting import figure, show
from bokeh.models import HoverTool

p = figure(plot_width=400, plot_height=400, tools='')
p.line(x=[5,4,3,2,1], y=[4,2,4,5,3])
p.add_tools(HoverTool(mode='vline'))
show(p)
```